File: src/main/java/pe/nanamochi/banchus/packets/client/MatchChangeModsPacket.java
package pe.nanamochi.banchus.packets.client;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.packets.Packet;
import pe.nanamochi.banchus.packets.Packets;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class MatchChangeModsPacket implements Packet {
  private int mods;

  @Override
  public Packets getPacketType() {
    return Packets.OSU_MATCH_CHANGE_MODS;
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/client/MatchChangeModsPacket.kt
package pe.nanamochi.banchus.packets.client

import java.io.InputStream
import pe.nanamochi.banchus.core.ClientPacket
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.io.DataReader

class MatchChangeModsPacket : ClientPacket() {
    override val type = PacketType.OSU_MATCH_CHANGE_MODS

    var mods: UInt = 0u

    override fun read(reader: DataReader, ins: InputStream) {
        this.mods = reader.readUint32(ins)
    }
}
